f5_openstack_agent/lbaasv2/drivers/bigip/agent_manager_lite.py

```diff
@@ -1652,3 +1652,92 @@ def rebuild_loadbalancer(self, context, loadbalancer, service):
                     "Fail to update status of loadbalancer %s "
                     "Exception: %s", lb_id, ex.message
                 )
+
+    @serialized('purge_loadbalancer')
+    @log_helpers.log_method_call
+    def purge_loadbalancer(self, context, loadbalancer, service):
+        lb_id = loadbalancer['id']
+        listeners = service.get("listeners", [])
+        pools = service.get('pools', [])
+        monitors = service.get("healthmonitors", [])
+        l7policies = service.get("l7policies", [])
+        members = service.get("members", [])
+
+        try:
+            bigip_device.set_bigips(service, self.conf)
+
+            # a l7policy contains l7rules, no need to rebuild l7rule.
+            for plc in l7policies:
+                plc_id = plc['id']
+                service["l7policy"] = plc
+
+                mgr = resource_manager.L7PolicyManager(self.lbdriver)
+                mgr.delete(plc, service)
+                LOG.debug("Finish to purge l7policy %s", plc_id)
+            LOG.debug("Finish to purge all l7policy in the lb")
+
+            for mn in monitors:
+                mn_id = mn['id']
+                service["healthmonitor"] = mn
+
+                mgr = resource_manager.MonitorManager(
+                    self.lbdriver, type=mn['type'])
+                mgr.delete(mn, service)
+                LOG.debug("Finish to purge monitor %s", mn_id)
+            LOG.debug("Finish to purge all monitors in the lb")
+
+            for mem in members:
```

Comment:
I remember that all pool members should be removed along with the pool, after invoke a deleting pool request, so that we might not need to purge those pool members separately one by one before purging the pool. Can you please try it in that way?